plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/utils.kt

```diff
@@ -232,6 +230,7 @@ internal fun KotlinType.replaceTypeArguments(newArguments: List<KSTypeArgument>)
     })
 
 internal fun FunctionDescriptor.toKSFunctionDeclaration(): KSFunctionDeclaration {
+    if (this.kind != CallableMemberDescriptor.Kind.DECLARATION) return KSFunctionDeclarationDescriptorImpl.getCached(this)
```

Comment:
Looks like it's time to fork `findPsi()`. We don't need the second part of the implementation.

Reply:
findPsi() itself isn't a problem in this case though, the underlying SourceElement is causing the issue.